site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+directly, which will be orders of magnitude more accurate than merely emulating
+its behaviour.
+
+## [Use deque instead of vector for the audio buffer](https://github.com/citra-emu/citra/pull/2958) by [Subv](https://github.com/Subv)
+
+Whenever the DSP consumes some frames from the audio buffer, Citra deletes them
+from it. This normally wouldn't pose any problems, but because the buffer was
+being stored as a vector, this led to some uneccessary operations. Namely, the
+C++ standard requires that all the data of a standard vector be in one contiguous
+block of memory. Because deleting frames from the buffer breaks this rule, Citra
+would automatically (1) allocate a new block of memory, (2) copy the entire buffer
+into that new block of memory, and (3) deallocate the old block of memory, thus
+deleting the old buffer.
+
+These steps are huge waste of time, as Citra doesn't need to guarantee that the
+audio buffer is in one contiguous block. So [Subv](https://github.com/Subv) changed
+the type of the buffer from a vector to a deque, which is essentially a queue that
+you can remove data from both the beginning and end of it. Because the contiguity
+requirement doesn't exist in deques, Citra doesn't do the uneccessary copying,
+leading to huge speed boosts in audio bound titles like Super Mario 3D Land, and
+even the Home Menu. Now, both run significantly faster!
+
+## [Add mingw64 compile support to appveyor](https://github.com/citra-emu/citra/pull/2912) by [jroweboy](https://github.com/jroweboy)
+
+When a program is written in a high-level programming language, such as C++, Rust,
+or Go, before the program can be run on a machine, it must be translated or "compiled"
+to machine code. Although it is possible to do this translation by hand, it is
+usually extremely difficult to do so and very time consuming. So instead, we have
+a program called a compiler than can automatically do this translation for us.
+
+This is also why a program compiled for an ARM machine cannot be run directly on
+an x86 machine, even when the source can work on either machine without issues.
+Instead this program must be translated, interpreted, or recompiled from source
+to x86. (In fact, this translation is exactly what [dynarmic](https://github.com/MerryMage/dynarmic)
+does to run code from a 3DS.)
+
+Every statement in a program must have an exact, unambiguous definition of what
+it does (its semantics). But, in the same way that a statement that means one
+thing can be written many different ways, and different compilers can translate the
+same statement many different ways.
+
+On Windows, there's two popular C++ compilers available as of today: MSVC++, which
+is the compiler Microsoft has written for Windows, and MINGW GCC, which is actually
+a port of the Linux `gcc` compiler to Windows. For better or worse, MINGW GCC
+compiles to a much more efficient binary than MSVC++, and so [jroweboy](https://github.com/jroweboy)
```

Comment:
I'm not sure this is really true (anymore)... I would soften this statement a little bit to avoid misinformation. Perhaps just "MINGW GCC has been shown to optimize Citra a little better than MSVC++"